text/0000-light-dom.md

```diff
@@ -0,0 +1,355 @@
+
+  Styles scoping in Light DOM will be done differently. The styles will be scoped not just to the component they belong, but also to its children. These styles won't leak out to the parents (or their children) though. This is similar to how Aura styles its components.
+
+  E.g.
+
+  ```html
+  <x-a>
+    <style>
+      p {
+        color: blue;
+      }
+    </style>
+
+    <p></p> <!-- will be blue -->
+    <p class="red"></p> <!-- will still be blue. red class from child won't leak out here -->
+
+    <x-b>
+      <style>
+        p.red {
+          color: red;
+        }
+      </style>
+
+      <p></p> <!-- will be blue. Style from x-a will be applied -->
+      <p class="red"></p> <!-- will be red -->
+      <p class="red"></p> <!-- This paragraph is coming from A and slotted into B. Will also be red. -->
+    </x-b>
+  </x-a>
+  ```
+
+  Opting out of this scoping is not supported. There's no way for a component author to say a CSS rule shouldn't be scoped to that specific component (and its children). For example, CSS modules provides a `:global` selector that can be used to opt out of this local scoping. LWC won't provide any such selector. If global scoping is desired, a global stylesheet can be injected manually.
+
+- **`this.template`**
+
+  In `LightningElement`, `this.template` returns the shadow-root. It will return `null` in `MacroElement`.
+
+### Security (WIP)
+
+- In some applications, light-dom components may not be allowed... it’s up to the app context
+  - **what is the behavior when it’s not allowed?**
+  - Some applications might disable light-dom as a whole
+  - Some applications might disable light-dom selectively using a “privileged code” model
+
+### Component Migration
+
+There is no migration of the existing components. Existing components inherit from `LightningElement` and light DOM components will inherit from `MacroElement` and they are two different things. If a component author wishes to use Light DOM for any of their existing components, they will have to make the relevant changes manually.
+
+### Synthetic Shadow DOM
+
+The selection of Light DOM should not be impacted by the use of synthetic shadow instead of native shadow. Now, the goal is to get rid of synthetic shadow, but this is hardly possible today because of:
+
+- Salesforce Lightning Components do not work with native shadow today
+- Shadow DOM has accessibility issues
+
+Could we think of a lightweight synthetic shadow that do not override the global methods but provides enough functions to the base components to work while letting third party integration tools work? This can be a different DOM option, which is an hybrid between Shadow DOM and Light DOM.
```

Comment:
I don't think this section belongs to the proposal. 
It overlaps more with @ravijayaramappa's work on Shadow DOM mixed mode.